### ISSUE TITLE
External url scheme is kept accross multiple url_for

### ISSUE THREAD

**Issue by paradoxxxzero** [pallets/flask#1663]
Consider this test case:

```python
from flask import Flask, url_for
app = Flask(__name__)


@app.route('/some/route')
def some_route():
    return 'Some route'


@app.route('/some/websocket')
def some_websocket():
    return 'Some websocket'


@app.route('/')
def test_case():
    return '<br><br>'.join((
        'some_route url:<br> %s' % url_for('some_route', _external=True),
        'some_websocket url:<br> %s' % url_for('some_websocket', _external=True, _scheme='ws'),
        'this test case url:<br> %s' % url_for('test_case', _external=True),
    ))

if __name__ == '__main__':
    app.config['SERVER_NAME'] = 'myserver.l:5000'
    app.run(debug=True)
```

Accessing http://myserver.l:5000/ gives:

```
some_route url:
http://myserver.l:5000/some/route

some_websocket url:
ws://myserver.l:5000/some/websocket

this test case url:
ws://myserver.l:5000/
```

should give:
```
some_route url:
http://myserver.l:5000/some/route

some_websocket url:
ws://myserver.l:5000/some/websocket

this test case url:
http://myserver.l:5000/
```

**Comment by ento** [comment]
also discussed here: #1596

**Comment by StephenOrJames** [comment]
This should be closed, it was fixed in #1813 by @mitsuhiko.
